website/docs/cdktf/concepts/resources.mdx

```diff
@@ -381,6 +381,236 @@ If you need to use the special [`self` object](/terraform/language/resources/pro
 If you need to ensure a condition is met either before or after a resource was created you can specify [conditions](/terraform/language/expressions/custom-conditions#preconditions-and-postconditions).
 To add one configure the `lifecycle` key on your resource with an object containing a `precondition` and / or a `postcondition`. These keys take a list of conditions with a `condition` key containing a Terraform Expression to be evaluated and an `errorMessage` key containing a string to be displayed if the condition is not met.
 
+## Importing Resources
+
+If you have existing resources that you want to manage with CDKTF, you can import them into your CDKTF application. The best way to do this is using the [`import` block feature](/terraform/language/import) of Terraform >= 1.5. You can do this in CDKTF either with a specified configuration or without, which will trigger the configuration generation.
+
+### Importing with Configuration
+
+To import a resource with configuration, you write your resource configuration as you would normally. Then you call the `importFrom` method on the resulting resource object. This method takes the resource ID as the first argument and the provider as an optional second. The provider is only required if you have multiple providers of the same type in your configuration.
```

Comment:
Upon reading this: Why don't we just take the provider that's passed to the resource and get rid of the second optional argument for `importFrom`? Would make it simpler for our users I think.